derive/src/js_enum_impl.rs

```diff
@@ -0,0 +1,77 @@
+extern crate proc_macro;
+
+use {proc_macro::TokenStream, quote::quote};
+
+pub fn js_enum_impl(input: TokenStream) -> TokenStream {
+    let ast: syn::ItemImpl = syn::parse(input).unwrap();
+    let ident = if let syn::Type::Path(path) = &*ast.self_ty {
+        path.path.get_ident().unwrap()
+    } else {
+        panic!("this ain't no enum impl")
+    };
+    let out_ident = quote::format_ident!("Js{}", ident);
+    let out_methods = ast
+        .items
+        .iter()
+        .filter_map(|x| {
+            if let syn::ImplItem::Method(method) = x {
+                let vis = &method.vis;
+                let sig = &method.sig;
+                let fn_name = &sig.ident;
+                let is_mut = method.sig.inputs.iter().find_map(|arg| {
+                    if let syn::FnArg::Receiver(arg) = arg {
+                        Some(arg)
+                    } else {
+                        None
+                    }
+                });
+                if is_mut.is_none() {
```

Comment:
You can safely delete this and add a `?` at the end of the `is_mut` expr instead (also, please, use a helper function for obtaining all of that)